src/python/pants/core/util_rules/environments.py

```diff
@@ -148,10 +165,11 @@ class DockerImageField(StringField):
     value: str
     help = softwrap(
         """
-        The docker image ID to use when this environment is loaded, e.g. `centos6:latest`.
+        The docker image ID to use when this environment is loaded.
 
-        TODO: expectations about what are valid IDs, e.g. if they must come from DockerHub vs other
-        registries.
+        This value may be any image identifier that the local Docker installation can accept.
+        This includes image names with or without tags (e.g. `centos6` or `centos6:latest`), or
+        image names with an immutable digest (e.g. `centos@sha256:<some_sha256_value>`).
```

Comment:
I didn't specify an actual sha256 value here, as I think these docs are probably going to be subject to bitrot (basically preserving bugs or vulnerabilities that are out of our control within our docs). I imagine people familiar with Docker will be able to figure out what a good value here would be.